### PR TITLE
Add support for kotlinx-train

### PR DESCRIPTION
**Pull request by dkhalanskyjb**

### CODE REVIEW

**Comment by hfhbd** [comment]
Just a general question: Why you don't use a Gradle settings plugin for the train setup to set up the repos and the versions but copy it over and over each kotlinx project? :)
This could also increase the usage of train in community projects/libraries.

**Comment by dkhalanskyjb** [comment]
Note that this is not a copy from another repository but a significantly cleaned-up version rewritten almost from scratch to play well with gradle.kts, in the other projects the train integrations are much more intrusive and behave differently. If this proposed integration does turn out to be correct, *maybe* it can be abstracted further into a plugin, but this is also not a given: a plugin that reads `../gradle.properties` is quite brittle, not a general-purpose solution by any means.

**Comment by anastasiiaSpaseeva** [comment]
Thanks for your notes, it's extracted to the task and planned to be researched in the upcoming sprint.

**Review by stasjas — APPROVED** [review]

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+val props = Properties().apply {
+    file("../gradle.properties").inputStream().use { load(it) }
+}
+
+// copy-pasted from `CommunityProjectsBuild`, see the explanation there
+fun RepositoryHandler.addTrainRepositories(project: Project) {
+    if (project.rootProject.properties["build_snapshot_train"]?.toString()?.toBoolean() == true) {
+        mavenLocal()
+        maven("https://oss.sonatype.org/content/repositories/snapshots")
```
[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply buildSrc/build.gradle.kts:19]: No idea, but the other projects have this as well. @qwwdfsad, who may know about this?

File: build.gradle.kts [hunk 2]
```diff
@@ -19,7 +19,9 @@ val modularJavaToolchainVersion by ext(project.property("java.modularToolchainVe
 
 allprojects {
     repositories {
+        addTrainRepositories(project)
         mavenCentral()
     }
 }
 
+logger.info("Using Kotlin compiler version: ${org.jetbrains.kotlin.config.KotlinCompilerVersion.VERSION}")
```
[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply build.gradle.kts:27]: What would it look like?

[the remark this answers is not in the document]
Comment (stasjas) [reply build.gradle.kts:27]: you can use 
freeCompilerArgs from kotlin options. Eg:
tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
    kotlinOptions.freeCompilerArgs += 
        "-version"
    kotlinOptions.apiVersion = kotlin_api_version
    kotlinOptions.languageVersion = kotlin_language_version
}

File: buildSrc/build.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply buildSrc/build.gradle.kts:19]: Probably a rudiment from the past.

Originally, some of our toolchain was using Sonatype snapshot repository for some of the publications

**Pull request merged**